### PR TITLE
Unify flash handling under flash_act

### PR DESCRIPTION
**Pull request by JohnWildkins**
I wanted to work on making explosions more performant, and that rework included adding the ability for explosions to have flashes. Then I looked into how to handle flashes, and was terrified. So I fixed some things.

changes:
  - balance: "All flash effects (flashbang, flashes, welding w.o protection, etc.) now deal damage to eyes relative to their effect intensity."
  - refactor: "Backend code for flash effects now unified under flash_act. Stunning sonic effects (e.g. flashbangs) now unified under soundbang_act."
  - refactor: "Significantly cleaned up handheld flash code."
  - tweak: "Vaurcae and Tajarans will now only take extra damage from flashes when their night vision is active, and when they do not have flash protection."
  - tweak: "Dionae now take flash damage and are weakened from all sources of flashes, rather than just handhelds. They also gain energy equal to the intensity of the flash."

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

**Comment by JohnWildkins** [comment]
Balance tweaks to handheld flashes:
 - They now are significantly more intense, being FLASH_PROTECTION_MAJOR (i.e. the same as welding without goggles)
 - They now enforce a cooldown of one use per second because of this power
 - Their description now shows how many charges remain
 - 'AOE' flash can now be activated by using it in-hand with harm intent
 - AOE flash only applies MAJOR flash to adjacent targets, MODERATE otherwise (so gigachads don't run around in a welding helmet stunning entire hallways)

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

File: code/game/gamemodes/cult/runes/stun.dm [hunk 1]
```diff
@@ -9,27 +9,22 @@
...
-			C.flash_eyes()
-			if(C.stuttering < 1 && NOT_FLAG(C.mutations, HULK))
-				C.stuttering = 1
-			C.Weaken(3)
-			C.confused = 10
-			C.Stun(3)
-			C.silent += 15
-			to_chat(C, SPAN_DANGER("The rune explodes in a bright flash!"))
-			admin_attack_log(user, C, "Used a stun rune.", "Was victim of a stun rune.", "used a stun rune on")
+			L.stuttering = TRUE
```
Comment (SleepyGemmy) [inline code/game/gamemodes/cult/runes/stun.dm:18]: this should be a integer, not a boolean.

**Review by SleepyGemmy — DISMISSED** [review]

File: code/game/gamemodes/cult/runes/stun.dm (hunk 1 above)
Comment (JohnWildkins) [reply code/game/gamemodes/cult/runes/stun.dm:18]: uh actually technically TRUE is just a define for 1 since BYOND has no Booleans so checkmate liberal 🤓

**Pull request merged**